session.go

```diff
@@ -647,9 +647,11 @@ func (s *session) GetGlobalSysVar(name string) (string, error) {
 func (s *session) SetGlobalSysVar(name string, value string) error {
 	if name == variable.SQLModeVar {
 		value = mysql.FormatSQLModeStr(value)
-		if _, err := mysql.GetSQLMode(value); err != nil {
+		sqlMode, err := mysql.GetSQLMode(value)
```

Comment:
We should not update the session when updating global.

Reply:
Evaluate the "final" server status each time when sending the packet?